src/current/v23.2/plpgsql.md

```diff
@@ -0,0 +1,417 @@
+- When specified as a parameter in a [`CREATE PROCEDURE`]({% link {{ page.version.version }}/create-procedure.md %}) statement, cursors can be accessed externally from the stored procedure.
+
+The cursor must first be opened within a PL/pgSQL block. If the cursor was declared without being bound to a query, you must specify a query using the `FOR` clause.
+
+~~~ sql
+BEGIN
+	OPEN cursor_name [ FOR query ];
+	...
+~~~
+
+You can then use the `FETCH` statement to retrieve query results for a specified number of rows:
+
+~~~ sql
+FETCH FORWARD rows FROM cursor_name;
+~~~
+
+You can free up a cursor variable by closing the cursor:
+
+~~~ sql
+CLOSE cursor_name;
+~~~
+
+Cursors that are specified as parameters, rather than declared as variables, can be passed externally to and from PL/pgSQL blocks. For example, using the [`movr` dataset]({% link {{ page.version.version }}/movr.md %}) loaded by [`cockroach demo`]({% link {{ page.version.version }}/cockroach-demo.md %}):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE PROCEDURE get_rides(rides_cursor REFCURSOR) AS $$
+  BEGIN
+    OPEN rides_cursor FOR SELECT * FROM movr.rides;
+  END
+  $$ LANGUAGE PLpgSQL;
+~~~
+
+The [`CALL`]({% link {{ page.version.version }}/call.md %}) and `FETCH` statement have to be issued within the same transaction, or the cursor will not be found:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BEGIN;
+  CALL get_rides('rides');
+  FETCH 2 FROM rides;
+  COMMIT;
+~~~
+
+~~~
+                   id                  |   city    | vehicle_city |               rider_id               |              vehicle_id              |         start_address         |         end_address         |     start_time      |      end_time       | revenue
+---------------------------------------+-----------+--------------+--------------------------------------+--------------------------------------+-------------------------------+-----------------------------+---------------------+---------------------+----------
+  ab020c49-ba5e-4800-8000-00000000014e | amsterdam | amsterdam    | b3333333-3333-4000-8000-000000000023 | bbbbbbbb-bbbb-4800-8000-00000000000b | 58875 Bell Ports              | 50164 William Glens         | 2018-12-16 03:04:05 | 2018-12-17 20:04:05 |   13.00
+  ab851eb8-51eb-4800-8000-00000000014f | amsterdam | amsterdam    | ae147ae1-47ae-4800-8000-000000000022 | bbbbbbbb-bbbb-4800-8000-00000000000b | 62025 Welch Alley             | 4092 Timothy Creek Apt. 39  | 2018-12-31 03:04:05 | 2019-01-02 03:04:05 |   32.00
+~~~
+
+### Report messages and handle exceptions
+
+Use the `RAISE` statement to print messages for status or error reporting.
+
+~~~ sql
+RAISE level 'message' [, expressions ]
+  [ USING option = 'expression' [, ... ] ];
+~~~
+
+- `level` is the message severity. Possible values are `DEBUG`, `LOG`, `NOTICE`, `INFO`, `WARNING`, and `EXCEPTION`. Specify `EXCEPTION` to raise an error that aborts the current transaction.
```

Comment:
It may be worth calling out here that RAISE currently only directly messages the client; it does not produce any output in the logs. Issue: https://github.com/cockroachdb/cockroach/issues/117750.